File: Java/Software Development Module - Java/Week12/PPS.java
public class PPS {
    
    //Data Members
    private String pps;
    private String result;

    //Constructor
    public PPS(String pps){
        this.pps = pps.toLowerCase();
    }

    public String computeAndGetVaidity(){

        boolean ppsCheck = true;

        if(this.pps.length() != 8){
            this.result = "Your PPS number is not the correct length";
            ppsCheck = false;
            return this.result;
        }
        if(ppsCheck){
            for(int i = 0; i < this.pps.length()-1;i++){

                if(!(Character.isDigit(pps.charAt(i)))){
                    ppsCheck=false;
                    this.result = "The first 7 characters of your PPS need to be Digits";
                    break;
                }

            }
        }

        if(ppsCheck){
            if(!(pps.charAt(pps.length()-1)=='a'||pps.charAt(pps.length()-1)=='e'||pps.charAt(pps.length()-1)=='i'||pps.charAt(pps.length()-1)=='o'||pps.charAt(pps.length()-1)=='u')){
                ppsCheck = false;
                this.result = "The last character of your PPS should be a Vowel";
            }
        }

        if(ppsCheck){
            this.result = "Your PPS is valid!";
        }

        return this.result;
    }

}

File: Python/Software Development Module - Python/Week12/PPS.py
"""
PPS.py
Billy Cussen
17/10/2020
"""

class myPPS:
    def __init__(self, pps):
        self.pps = pps

    def checkPPS(self):
        if len(self.pps) != 8:
            return "This is not a valid PPS, PPS Numbers have 8 characters (e.g. 1234567U)"
        elif self.pps[:7].isdigit() == False:
            return "This is not a valid PPS, the first 7 characters of a PPS should be Digits (e.g. 1234567U)"
        elif self.pps[7].isalpha() == False:
            return "This is not a valid PPS, the last character of a PPS Number should be a Number (e.g. 1234567U)"
        else:
            return "This is a valid PPS Number"
         